cider/tests/functionality/DateTypeQueryTest.cpp

```diff
@@ -247,9 +247,42 @@ TEST_F(DateRandomAndNullQueryTest, DateOpTest) {
       "< date '1980-01-01'");
 }
 
+class TimeTypeQueryTest : public CiderTestBase {
+ public:
+  TimeTypeQueryTest() {
+    table_name_ = "test";
+    create_ddl_ = "CREATE TABLE test(col_a DATE, col_b TIME, col_c TIMESTAMP);";
+    input_ = {std::make_shared<CiderBatch>(
+        QueryDataGenerator::generateBatchByTypes(100,
+                                                 {"col_a", "col_b", "col_c"},
+                                                 {CREATE_SUBSTRAIT_TYPE(Date),
+                                                  CREATE_SUBSTRAIT_TYPE(Time),
+                                                  CREATE_SUBSTRAIT_TYPE(Timestamp)},
+                                                 {2, 2, 2},
+                                                 GeneratePattern::Random))};
+  }
+};
+
+TEST_F(TimeTypeQueryTest, TimeStampTest) {
+  assertQuery("SELECT col_c +  interval '1' month  from test",
+              "add_timestamp_interval_month.json");
+  assertQuery("SELECT col_c +  interval '1' day  from test",
+              "add_timestamp_interval_day.json");
+  assertQuery("select extract(microsecond from col_c) from test",
+              "extract/microsecond_of_timestamp.json");
+  assertQuery("select cast(col_a as TIMESTAMP) from test", "cast_date_as_timestamp.json");
+  // equals to date trunc
+  assertQuery("SELECT cast(col_c as DATE) from test", "cast_timestamp_as_date.json");
+  assertQuery("select extract(second from col_b) from test",
```

Comment:
did cider support query like `SELECT col_b + interval (1) second`?

Reply:
As https://github.com/substrait-io/substrait/blob/main/extensions/functions_datetime.yaml#L77 substrait plan is not yet supported. but cider is already supported add_interval_seconds function,  have added new related test.